src/transactions/management/transactionWorker.ts

```diff
@@ -22,9 +23,38 @@ import { InputGenerateTransactionOptions, InputGenerateTransactionPayloadData, S
 import { AccountSequenceNumber } from "./accountSequenceNumber";
 import { AsyncQueue, AsyncQueueCancelledError } from "./asyncQueue";
 
-const promiseFulfilledStatus = "fulfilled";
+export const promiseFulfilledStatus = "fulfilled";
 
-export class TransactionWorker {
+export enum TransactionWorkerEventsEnum {
+  TransactionSent = "transactionSent",
+  TransactionSendFailed = "transactionSendFailed",
+  TransactionExecuted = "transactionExecuted",
+  TransactionExecutionFailed = "transactionExecutionFailed",
+  ExecutionFinish = "executionFinish",
+}
+
+export interface TransactionWorkerEvents {
+  transactionSent: (data: SuccessEventData) => void;
+  transactionSendFailed: (data: FailureEventData) => void;
+  transactionExecuted: (data: SuccessEventData) => void;
+  transactionExecutionFailed: (data: FailureEventData) => void;
+  executionFinish: (data: ExecutionFinishEventData) => void;
+}
```

Comment:
Ok, just double checking I understand each event:
- `transactionSent` is fired after a transaction gets sent to the chain
- `transactionSendFailed` is fired if there is an error sending the transaction to the chain
- `transactionExecuted` is fired when a single transaction has run successfully
- `transactionExecutionFailed` is run if a single transaction fails in execution
- `executionFinish` is run when the queue has been emptied

So if someone wanted to make an async function to batch requests, it would look like this:


```ts
// Pseudo code, returns txn_hash array
type TxnHash = string;
function batchTransactionExample(sender: Account, payload: InputGenerateTransactionPayloadData[]): Promise<{
	succeeded_txns: TxnHash[],
	failed_txns: TxnHash[]
}> {
	return new Promise(async (resolve, reject) => {
		let succeeded_txns = [];
		let failed_txns = [];
		
		aptos.transaction.batch.forSingleAccount({sender, data: payload})

		// Handle transaction successes, we don't care about sent, just executed
		aptos.transaction.batch.on(TransactionWorkerEvents.TransactionExecuted, async (data: any) => {
			succeeded_txns.push(data.txn_hash);
		});

		// Handle transaction failures (execution or send)
		aptos.transaction.batch.on(TransactionWorkerEvents.TransactionExecutionFailed, async (data: any) => {
			failed_txns.push(data.txn_hash);
		});
		aptos.transaction.batch.on(TransactionWorkerEvents.TransactionSendFailed, async (data: any) => {
			failed_txns.push(data.txn_hash);
		});

		// Handle resolving the promise
		aptos.transaction.batch.on(TransactionWorkerEvents.ExecutionFinish, async (data: any) => {
			resolve({ succeeded_txns, failed_txns })
		});
	})
}
```
I'm guessing the caller will have to unsubscribe from the emitter? That is something we would want to highlight in the two examples potentially.

Reply:
yep! I'll add some doc comments - good catch!